docs/examples/piecewise.py

```diff
@@ -0,0 +1,44 @@
+"""
+
+Piecewise-constant coefficients.
+
+"""
+
+from skfem import *
+import numpy as np
+
+lengths = [3, 2]
+
+mesh = (MeshLine(np.linspace(-lengths[0], lengths[1], 6))
+        * MeshLine(np.linspace(0, 1, 2)))._splitquads()
+basis = InteriorBasis(mesh, ElementTriP1())
+basis0 = InteriorBasis(mesh, ElementTriP0())
+
+
+@bilinear_form
+def conduction(u, du, v, dv, w):
+    return w.w * sum(du * dv)
+
+
+k = np.ones(basis.N) * lengths[1]
+k[mesh.elements_satisfying(lambda x, y: x < 0)] = lengths[0]
+L = asm(conduction, basis, w=basis0.interpolate(k))
```

Comment:
```suggestion
X, Y = basis.global_coordinates()
k = 3 * (X < 0) + 2 * (X > 0) 
L = asm(conduction, basis, w=k)
```
This should also work.

Reply:
Note: I should've read your issue more carefully. This approach is not as general as you probably require if you eventually want to use Submeshes to define k.